flink-core/src/main/java/org/apache/flink/util/NetUtils.java

```diff
@@ -196,4 +198,35 @@ public static String hostAndPortToUrlString(String host, int port) throws Unknow
 		}
 		return finalSet;
 	}
+
+	/**
+	 * Tries to allocate a socket from the given sets of ports.
+	 *
+	 * @param portsSet A set of ports to choose from. The method will remove the ports from the set!
+	 * @param factory A factory for creating the SocketServer
+	 * @param logger Logger instance
+	 * @return null if no port was available or an allocated socket.
+	 */
+	public static ServerSocket createSocketFromPorts(Set<Integer> portsSet, SocketFactory factory, Logger logger) throws IOException {
```

Comment:
Why do you pass the logger as a parameter to the method? That way, it is really hard to know from where the actual logging call was executed.